Capabilities status blog post (#2172)
GoogleChrome/web.dev: src/site/content/en/blog/fugu-status/index.md

---

> title: New Capabilities Status

Reviewer: ```suggestion
title: New capabilities status
```

Reviewer: Sentence case in titles.

(I've never read an explanation for why Google favors this. Historically, different western languages would have different rules for title capitalization. The English rules are somewhat technical, if you remember from school. It occurs to me that sentence case is easily translatable to most if not all western languages.)

capabilities of native platforms to the web platform, while maintaining user
security, privacy, trust, and other core tenets of the web.

You can see the full list of new and potential capabilities at
<https://goo.gle/fugu-api-tracker>.

## Capabilities available behind a flag {: #flag }

These APIs are only available behind a flag because they're still under
development and should be considered *experimental*. There's a good chance
they'll break, or the API surface will change.

<div class="w-table-wrapper">
  <table>
    <thead>
      <tr>
        <th>Capability</th>
        <th>Description</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <a href="/shape-detection/">Shape&nbsp;Detection API</a>
        </td>
        <td>
          The Shape Detection API opens up native implementations of shape
          detection services and exposes them through a set of JavaScript
          interfaces.
        </td>
      </tr>
    </tbody>
  </table>
</div>

## Capabilities available as an origin trial {: #origin-trial }

These APIs are available as an [origin trial][ot-dashboard] in Chrome. Origin

> trials provide an opportunity for us to validate experimental features and

Reviewer: ```suggestion
trials provide an opportunity for Chrome to validate experimental features and
```

APIs, and make it possible for you to provide feedback on their usability

> and effectiveness in  broader deployment.

Reviewer: ```suggestion
and effectiveness in broader deployment.
```

          application, such as the shelf or home screen. Badging makes it easy
          to subtly notify the user that there is some new activity that might
          require their attention, or it can be used to indicate a small
          amount of information, such as an unread count.
        </td>
      </tr>
      <tr>
        <td>
          <a href="/content-indexing-api/">Content Indexing API</a>
        </td>
        <td>
          Your PWA might cache articles and media files, but how will your users
          know that your pages work while offline? The Content Indexing API is
          one answer to this question currently in an origin trial. Once the
          index is populated with content from your PWA, as well as any other
          installed PWAs, it will show up in dedicated areas of supported
          browsers.
        </td>
      </tr>
      <tr>
        <td>
          <a href="/native-file-system/">
            Native File System API
          </a>
        </td>
        <td>
          The Native File System API (formerly known as the Writable Files API)
          enables developers to build powerful web apps that interact with files

>           on the users local device, like IDEs, photo and video editors, text

Reviewer: ```suggestion
          on the user's local device, like IDEs, photo and video editors, text
```

          editors, and more. After a user grants a web app access, this API
          allows web apps to read or save changes directly to files and folders

>           on the users device.

Reviewer: ```suggestion
          on the user's device.
```

      <tr>
        <td>Notification Triggers</td>
        <td>
          Notification Triggers let you schedule notifications in advance, so
          that the operating system will deliver the notification at the right
          time - even if there is no network connectivity, or the device is in
          battery saver mode.
        </td>
      </tr>
      <tr>
        <td>Serial API</td>
        <td>
          The Web Serial API provides a way for websites to read from and
          write to a serial device with scripts. The API bridges the web and
          the physical world by allowing websites to communicate with serial
          devices, such as microcontrollers and 3D printers.
        </td>
      </tr>
      <tr>
        <td>
          <a href="/sms-receiver-api-announcement/">SMS Receiver API</a>
        </td>
        <td>
          Finding, memorizing, and typing OTPs sent via SMS is cumbersome.
          The SMS Receiver API simplifies the OTP workflow for users.
        </td>
      </tr>
      <tr>
        <td>
          <a href="/wakelock/">Wake Lock API</a>
        </td>
        <td>
          To avoid draining the battery, most devices will quickly fall asleep

>           when left idle. While this is fine for most of the time, there are

Reviewer: ```suggestion
          when left idle. While this is fine most of the time, there are
```

          some applications that need to keep the screen or the device awake in
          order to complete some work. The Wake Lock API provides a way to
          prevent the device from dimming or locking the screen or prevent
          the device from going to sleep when an application needs to keep
          running.
        </td>
      </tr>
    </tbody>
  </table>
</div>

## Capabilities available in stable {: #in-stable }

The following APIs have graduated from origin trial and are available in the

> latest version of Chrome, and in many cases other Chromium based browsers.

Reviewer: (Aside: We should list the other browsers' availability.)

Author: MDN or CanIUse? What would you suggest for a place to link. Ideally I don't want to have to test this myself and update it every release.

        </td>
        <td>
          Periodic Background Sync enables web applications to periodically
          synchronize data in the background, bringing web apps closer to
          the behavior of a native app.
        </td>
      </tr>
      <tr>
        <td>
          <a href="/web-share/">Web Share</a>
        </td>
        <td>
          With the Web Share API, web apps are able to use the same
          system-provided share capabilities as native apps. The Web Share API
          makes it possible for web apps to share links, text, and files to
          other apps installed on the device in the same way as native apps.
        </td>
      </tr>
      <tr>
        <td>
          <a href="/web-share-target/">Web Share Target</a>
        </td>
        <td>
          The Web Share Target API allows installed web apps to register with
          the underlying OS as a share target to receive shared content from
          either the Web Share API or system events, like the OS-level share
          button.
        </td>
      </tr>
    </tbody>
  </table>
</div>

## Suggest a new capability {: #suggest-new }

> Have a suggestion for a capability you think we should consider?

Reviewer: ```suggestion
Do you have a suggestion for a capability you think we should consider?
```

---
title: New capabilities status

> subhead: Web apps should be able to do anything native apps can. We want to make it possible for you to build and deliver apps on the open web that have never been possible before.

Reviewer 2: ```suggestion
subhead: Web apps should be able to do anything native apps can. Google wants to make it possible for you to build and deliver apps on the open web that have never been possible before.
```

Reviewer 2: I'm using "Google" here rather than "Chrome" because the first paragraph of the post says the Fugu is a "cross-company effort"

date: 2018-11-12
updated: 2020-02-19
tags:
  - post
  - capabilities
  - fugu
draft: true
---

{% Aside %}
This is a living document and will be updated as features move between
states.
{% endAside %}

> The [capabilities project][capabilities-project] is a cross-company effort

Reviewer 2: ```suggestion
The [capabilities project][capabilities-project] is a cross-company effort at Google
```

to make it possible for web apps to do anything native apps can, by exposing the
capabilities of native platforms to the web platform, while maintaining user
security, privacy, trust, and other core tenets of the web.

You can see the full list of new and potential capabilities at

> <https://goo.gle/fugu-api-tracker>.

Reviewer 2: ```suggestion
the [Fugu API Tracker](https://goo.gle/fugu-api-tracker).
```

        </td>
        <td>
          Periodic Background Sync enables web applications to periodically
          synchronize data in the background, bringing web apps closer to
          the behavior of a native app.
        </td>
      </tr>
      <tr>
        <td>
          <a href="/web-share/">Web Share</a>
        </td>
        <td>
          With the Web Share API, web apps are able to use the same
          system-provided share capabilities as native apps. The Web Share API
          makes it possible for web apps to share links, text, and files to
          other apps installed on the device in the same way as native apps.
        </td>
      </tr>
      <tr>
        <td>
          <a href="/web-share-target/">Web Share Target</a>
        </td>
        <td>
          The Web Share Target API allows installed web apps to register with
          the underlying OS as a share target to receive shared content from
          either the Web Share API or system events, like the OS-level share
          button.
        </td>
      </tr>
    </tbody>
  </table>
</div>

## Suggest a new capability {: #suggest-new }

> Do you have a suggestion for a capability you think we should consider?

Reviewer 2: ```suggestion
Do you have a suggestion for a capability you think Chrome should consider?
```
